packages/grafana-ui/src/components/Chart/TooltipContainer.tsx

```diff
@@ -20,18 +20,37 @@ export const TooltipContainer: React.FC<TooltipContainerProps> = ({
 }) => {
   const theme = useTheme();
   const tooltipRef = useRef<HTMLDivElement>(null);
+  const tooltipMeasurementRef = useRef<Dimensions2D>({ width: 0, height: 0 });
   const { width, height } = useWindowSize();
   const [placement, setPlacement] = useState({
     x: positionX + offsetX,
     y: positionY + offsetY,
   });
 
+  const resizeObserver = new ResizeObserver((entries) => {
+    for (let entry of entries) {
+      const tW = Math.floor(entry.contentRect.width);
+      const tH = Math.floor(entry.contentRect.height);
```

Comment:
`contentRect` does not include the 8px padding, maybe add this as fudge here until Safari supports `borderBoxSize`